integration-tests/test_client.py

```diff
@@ -0,0 +1,84 @@
+import subprocess
+import pytest
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_client_files_permission(insights_client):
+    """Verify that permission for lastupload file is 0644:
+     file /etc/insights-client/.lastupload
+    Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1924990
+    """
+    file_last_upload = "/etc/insights-client/.lastupload"
+    with contextlib.suppress(FileNotFoundError):
+        os.remove(file_last_upload)  # performing a cleanup before test
+    insights_client.register()
+    assert insights_client.is_registered
+    assert os.stat(file_last_upload).st_mode == 0o100644
+
+
+@pytest.fixture()
+def rpm_ql_insights_client():
+    cmd = ["rpm", "-ql", "insights-client"]
+    list_file = subprocess.check_output(cmd, universal_newlines=True)
+    return list_file.split("\n")
+
+
+@pytest.mark.parametrize(
+    "file",
+    [
+        "/etc/insights-client/insights-client.motd",
+        "/etc/insights-client/insights-client.conf",
+        "/usr/lib/systemd/system/insights-client-boot.service",
+        "/usr/lib/systemd/system/insights-client-results.path",
+        "/usr/lib/systemd/system/insights-client-results.service",
+        "/usr/lib/systemd/system/insights-client.service",
+        "/usr/lib/systemd/system/insights-client.timer",
+        "/usr/share/doc/insights-client",
+        "/usr/share/doc/insights-client/file-content-redaction.yaml.example",
+        "/usr/share/doc/insights-client/file-redaction.yaml.example",
+        "/usr/share/man/man5/insights-client.conf.5.gz",
+        "/usr/share/man/man8/insights-client.8.gz",
+        "/etc/logrotate.d/insights-client",
+    ],
+)
+def test_client_rpm_mandatory_files(filename, rpm_ql_insights_client):
+    """
+    Verify the existence of mandatory files for insights-client rpm
+    """
+    assert (
+        filename in rpm_ql_insights_client
+    ), f"{filename} is not in insights-client package"
+
+
+def test_client_logfiles_mask(insights_client):
+    """Verify that files in /var/log/insights-client have the right mode: 0600
+
+    Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1955724
+    """
+    logfiles = glob.glob("/var/log/insights-client/*.log*")
+    # It is necessary to perform some command using insights-client
+    # to populate logs
+    insights_client.register()
+    for logfile in logfiles:
+        permissions = os.stat(logfile).st_mode
+        assert (
+            permissions == 0o100600
+        ), f"permission on file {logfile} must be: 600, current it is: {permissions}"
+
+
+def test_client_logdir_permissions():
+    """Verify that permissions on directory /var/log/insights-client
+    have the right mode: 0700
+    """
+    logdir_name = "/var/log/insights-client"
+    permissions = os.stat(logdir_name).st_mode
+    assert (
+        permissions == 0o40700
+    ), f"permissions on dir {logdir_name} must be: 0700, current one is: {permissions}"
```

Comment:
this label for `assert` is very much redundant: on case of failure, pytest will print the "expected" and "got" values already; also, this test checks exactly one thing, so it is clear that, if it fails, that specific directory has wrong permissions